packages/dds/tree/src/feature-libraries/editable-tree/editableTreeSequence.ts

```diff
@@ -0,0 +1,335 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ITreeSubscriptionCursor, TreeNavigationResult } from "../../forest";
+import { NamedTreeSchema, TreeSchemaIdentifier, lookupTreeSchema } from "../../schema-stored";
+import { Anchor, FieldKey } from "../../tree";
+import {
+    EditableTree,
+    EditableTreeOrPrimitive,
+    getTypeSymbol,
+    inProxyOrUnwrap,
+    ProxyTarget,
+    proxyTargetSymbol,
+    UnwrappedEditableField,
+} from "./editableTree";
+import { ProxyContext } from "./editableTreeContext";
+import { AdaptingProxyHandler, getArrayOwnKeys, keyIsValidIndex } from "./utilities";
+
+/**
+ * Sequence field nodes as a sequence of EditableTrees or primitive values (if unwrapped).
```

Comment:
If you get an EditableTreeSequence by iterating an EditiableTree node, it might not correspond to a sequence field: you could get one for some other type of field like a value field.

Since all our fields have a schema and field key (including the root, though its field key is a detached sequence id, that still can be used as a LocalFieldKey for it), I think it might make sense to add a FieldSchema and FieldKey fields to this type, and rename it `EditableField` (replacing the existing EditiableField type).

I think that would help simplify the API, while fixing the name of this type at the same time.